ml_genn/ml_genn/compilers/event_prop_compiler.py

```diff
@@ -909,44 +918,83 @@ def build_neuron_model(self, pop: Population, model: NeuronModel,
-                            dynamics=dynamics_code,
-                            transition=transition_code))
-
-                    # Prepend (as it accesses the pre-reset value of V) 
-                    # code to reset to write spike time and I-V to ring buffer
-                    model_copy.prepend_reset_code(
-                        neuron_reset.substitute(
-                            max_spikes=self.max_spikes,
-                            write="RingIMinusV[ringOffset + RingWriteOffset] = (Isyn * IsynScale) - V;",
-                            strict_check=(neuron_reset_strict_check 
-                                        if self.strict_buffer_checking
-                                        else "")))
-                    
                     # Add custom updates to calculate softmax from TFirstSpike
                     compile_state.batch_softmax_populations.append(
                         (pop, "TFirstSpike", "Softmax"))
+                elif rmse_loss:
+                    # Add parameters to model to hold delta
+                    model_copy.add_param("Delta", "scalar", 
+                                         compile_state.losses[pop].delta)
+                    
+                    # Add state variable to hold sum of TFirstSpike from previous trial
+                    # **YUCK** REALLY should be timepoint but then you can't softmax
+                    model_copy.add_var("TFirstSpikeSumBack", "scalar", 0.0,
+                                       VarAccess.READ_ONLY_SHARED_NEURON)
+                    
+                    # Add another variable to hold TFirstSpike for the correct 
+                    # output neuron from the previous trial at which
+                    # **YUCK** REALLY should be timepoint but then you can't softmax
+                    model_copy.add_var("TFirstSpikeTrueBack", "scalar", 0.0, 
+                                       VarAccess.READ_ONLY_SHARED_NEURON)
+
+                    # Add population to list requiring bespoke TTFS reduction
+                    compile_state.ttfs_reduce_populations.append(pop)
+                    
+                    # In backward pass, update lambdas 
+                    dynamics_code = f"""
+                        LambdaI = (A * LambdaV * (Alpha - Beta)) + (LambdaI * Beta);
+                        LambdaV *= Alpha;
+                        """
+                
+                    # On backward pass transition, update LambdaV if this is the first spike on an incorrect neuron
+                    transition_code = f"""
+                        if (fabs(backT + TFirstSpikeBack) < 1e-3*dt) {{
+                            const scalar iMinusVRecip = 1.0 / RingIMinusV[ringOffset + RingReadOffset];
+                            if(id == YTrueBack) {{
+                                LambdaV += (TFirstSpikeSumBack - (num_neurons * TFirstSpikeBack) + ((num_neurons - 1) * Delta)) * iMinusVRecip;
```

Comment:
would it be clearer if ```TFirstSpikeTrueBack``` was used (even though it is of course the same value in this branch of the if clause)?